src/classes/speciessite.cpp

```diff
@@ -444,9 +445,35 @@ const std::vector<std::vector<int>> &SpeciesSite::sitesXAxisAtomsIndices() const
 const std::vector<std::vector<int>> &SpeciesSite::sitesYAxisAtomsIndices() const { return sitesYAxisAtomsIndices_; }
 
 /*
- * Generation from Parent
+ * Generation
  */
 
+// Calculate geometric centre of atoms in the given molecule
+Vec3<double> SpeciesSite::centreOfGeometry(std::vector<int> &indices) const
+{
+    const auto ref = parent_->atom(indices.front()).r();
+    return std::accumulate(std::next(indices.begin()), indices.end(), ref,
+                           [&ref, this](const auto &acc, const auto idx)
+                           { return acc + parent_->box()->minimumImage(parent_->atom(idx).r(), ref); }) /
+           indices.size();
+}
+
+// Calculate (mass-weighted) coordinate centre of atoms in the given molecule
+Vec3<double> SpeciesSite::centreOfMass(std::vector<int> &indices) const
+{
+    auto mass = AtomicMass::mass(parent_->atom(indices.front()).Z());
+    const auto ref = parent_->atom(indices.front()).r();
+    auto sums = std::accumulate(std::next(indices.begin()), indices.end(), std::pair<Vec3<double>, double>(ref * mass, mass),
+                                [&ref, this](const auto &acc, const auto idx)
+                                {
+                                    auto mass = AtomicMass::mass(parent_->atom(idx).Z());
+                                    return std::pair<Vec3<double>, double>(
+                                        acc.first + parent_->box()->minimumImage(parent_->atom(idx).r(), ref) * mass,
+                                        acc.second + mass);
```

Comment:
It might be that C++17 won't let this happen, but can we do this instead?
```suggestion
                                    return {acc.first + parent_->box()->minimumImage(parent_->atom(idx).r(), ref) * mass,
                                        acc.second + mass};
```